src/main/java/com/mycompany/myapp/service/reservation/ReservationService.java

```diff
@@ -51,15 +50,27 @@ public void deleteReservation(Long id) {
 
     public List<Reservation> getAllByPoolId(String poolId) {
         Pool pool = poolRepository.getFullByIdWithReservation(poolId);
+        if (pool == null) {
```

Comment:
Is it even possible to get null there ?

Reply:
Yes, when the pool doesn't exist yet in database, for example we upload updated csv with some new machines.

Reply:
ok then